nevergrad/optimization/mutations.py

```diff
@@ -69,7 +71,19 @@ def portfolio_discrete_mutation(self, parent: tp.ArrayLike, intensity: tp.Option
             boolean_vector = [self.random_state.rand() > (float(intensity) / dimension) for _ in parent]
         return [s if b else self.significantly_mutate(s, arity) for (b, s) in zip(boolean_vector, parent)]
 
+    def coordinatewise_mutation(self, parent: tp.ArrayLike, velocity: tp.ArrayLike, boolean_vector: tp.ArrayLike, arity: int) -> tp.ArrayLike:
+        """This is the anisotropic counterpart of the classical 1+1 mutations in discrete domains
+        with tunable intensity: it is useful for anisotropic adaptivity."""
+        dimension = len(parent)
+        boolean_vector = np.zeros(dimension, dtype=bool)
+        while not any(boolean_vector):
+            boolean_vector = self.random_state.rand(dimension) < (1. / dimension)
+        discrete_data = discretization.threshold_discretization(parent, arity=arity)
+        return discretization.inverse_threshold_discretization([s if not b else s + np.random.choice([-1. ,1.]) * v for (b, s, v) 
```

Comment:
can you please vectorize this?
something like:
discrete_data[boolean_vector] += np.random.choice([-1, 1], size=sum(boolean_vector))

Reply:
Done!